aiohttp/http_parser.py

```diff
@@ -505,6 +517,9 @@ def parse_headers(
                 close_conn = False
             # https://www.rfc-editor.org/rfc/rfc9110.html#name-101-switching-protocols
             elif v == "upgrade" and headers.get(hdrs.UPGRADE):
```

Comment:
Isn't this the source of the problem for #6446?  Just because the `Upgrade` header has content, doesn't mean there will be an upgrade.  The server has every right to ignore it if the scheme is unsupported.

Reply:
If you think you can provide a fix, that'd be great. I believe there is already an xfail test from that issue, so as long as you can get that test passing..

Reply:
One problem at a time for this newbie... Brotli first 😉